packages/aleo-gateway/src/weighted_signers.rs

```diff
@@ -44,35 +56,104 @@ impl TryFrom<&VerifierSet> for WeightedSigners {
             .collect::<Result<Vec<_>, _>>()?;
 
-        signers.sort_by(|signer1, signer2| signer1.signer.cmp(&signer2.signer));
+        // signers.sort_by(|signer1, signer2| signer1.signer.cmp(&signer2.signer));
+        signers.sort_by(|signer1, signer2| {
+            /* give the lowest priority to the default address */
+            if signer1.signer == Address::default() {
+                std::cmp::Ordering::Greater
+            } else if signer2.signer == Address::default() {
+                std::cmp::Ordering::Less
+            } else {
+                signer1.signer.cmp(&signer2.signer)
+            }
+        });
 
         let threshold = value.threshold;
-        let nonce = [0, 0, 0, value.created_at];
+        let _nonce = [0, 0, 0, value.created_at];
+
+        // Distribute signers across groups
+        let mut grouped_signers = Vec::with_capacity(GROUPS);
+        for group_idx in 0..GROUPS {
+            let start = group_idx * GROUP_SIZE;
+            let end = start + GROUP_SIZE;
+
+            let group_vec: Vec<_> = signers[start..end].to_vec();
+            let group_array: [WeightedSigner; GROUP_SIZE] = group_vec.try_into().expect(&format!(
+                "Group {} should have exactly {} elements",
+                group_idx, GROUP_SIZE
+            ));
+
+            grouped_signers.push(group_array);
+        }
+
+        // Convert Vec<[WeightedSigner; GROUP_SIZE]> to [[WeightedSigner; GROUP_SIZE]; GROUPS]
+        let signers_array: [[WeightedSigner; GROUP_SIZE]; GROUPS] = grouped_signers
+            .try_into()
+            .unwrap_or_else(|_| panic!("Failed to convert to array of size {}", GROUPS));
 
         Ok(WeightedSigners {
-            signers,
+            signers: signers_array,
             threshold,
-            nonce,
+            // nonce,
         })
     }
 }
 
-impl AleoValue for WeightedSigners {
+impl<const GROUP_SIZE: usize, const GROUPS: usize> AleoValue
+    for WeightedSigners<GROUP_SIZE, GROUPS>
+{
     fn to_aleo_string(&self) -> Result<String, Report<Error>> {
-        let signers = self
-            .signers
-            .iter()
-            .map(WeightedSigner::to_aleo_string)
-            .collect::<Result<Vec<_>, Report<Error>>>()?
-            .join(", ");
-        let res = format!(
-            r#"{{ signers: [ {} ], threshold: {}u128, nonce: [ {}u64, {}u64, {}u64, {}u64 ] }}"#,
-            signers, self.threshold, self.nonce[0], self.nonce[1], self.nonce[2], self.nonce[3]
-        );
+        // Start with the opening part of the string
+        let mut res = String::from("{ signers: [ ");
+
+        // Add each group's formatted string
+        for (i, group) in self.signers.iter().enumerate() {
+            let group_str = format!(
+                "[{}]",
+                group
+                    .iter()
+                    .map(|s| s.to_aleo_string())
+                    .collect::<Result<Vec<_>, _>>()?
+                    .join(", ")
+            );
```

Comment:
As per my other comment, I suggest to write directly to the `String` buffer (like the rest of the code is doing) rather than calling `format!` in a loop, since it will allocate memory for a new `String` each time.